### ISSUE TITLE
Image thumbnails are not displayed

### ISSUE THREAD

**Issue by MrFoxyGmFr** [filebrowser/filebrowser#2078]
I recently encountered a problem that some photos don't display thumbnails.

![bug](https://user-images.githubusercontent.com/33060302/182663940-e7431dbb-49cf-4436-b088-3f1e15b8ad24.png)

While researching this problem in more detail, I noticed that photos that don't display a thumbnail have these lines in their exif:
```
Thumbnail Offset                : 2048
Thumbnail Length                : 0
```

The point is that the logic of the current imageService does not imply the existence of the Thumbnail Length exif field with the value 0, it returns an empty thumbnail, as if it should be. 
[Proof](https://github.com/filebrowser/filebrowser/blob/aaed985699b3c63092ecb02c8bc07634123360ab/img/service.go#L158) &mdash; will return thm if errThm == nil. In this particular situation thm is an empty byte fragment, errThm is nil. The problem is that neither the getEmbeddedThumbnail function nor this if construction implies that thm can be empty.

After conducting an experiment, I realized that this error could be corrected by adding one more statement to this construction.
My implementation is shown below, it successfully fixed my thumbnail problem.

```go
if config.quality == QualityLow && format == FormatJpeg {
	thm, newWrappedReader, errThm := getEmbeddedThumbnail(wrappedReader)
	wrappedReader = newWrappedReader
	if errThm == nil && len(thm) != 0 {
		_, err = out.Write(thm)
		if err == nil {
			return nil
		}
	}
}
```

I decided not to create a PR for such a minor problem, so I am writing this issue expecting that the above will improve filebrowser and make it even more usable.

**Comment by ramiresviana** [comment]
Can you provide server environment details, configuration, server logs and file samples?

**Comment by MrFoxyGmFr** [comment]
I am using the official docker image with a configuration very similar to the standard one.
There is nothing in the logs that expands on the problem.

Here is an example of files and how they are displayed (these files differ only in EXIF):

![Screenshot 2022-09-28 at 15 09 06](https://user-images.githubusercontent.com/33060302/192775057-d425155f-e08b-4945-9043-85c8f426b2a2.png)

The file "with_thumbnail.jpg" contains the above two lines, but the file "without_thumbnail.jpg" does not.

Files: [with_thumbnail.jpg](https://user-images.githubusercontent.com/33060302/192775486-d2a2891e-b8d1-412e-963c-468239ced46d.jpg), [without_thumbnail.jpg](https://user-images.githubusercontent.com/33060302/192775524-dfae198b-4a9d-4498-b434-fca60d9c56b0.jpg)

As I wrote above, I have already found the place where the mistake is most likely made and gave my solution.
Hope for a speedy bug fix.

**Comment by MrFoxyGmFr** [comment]
Any information on the current status of the issue?

**Comment by MrFoxyGmFr** [comment]
?

**Comment by jerrywoo96** [comment]
up

**Comment by z1haze** [comment]
lol, nice. I too would like to understand how to get thumbnails for image files in the filebrowser

**Comment by SaltyRiver** [comment]
@z1haze @MrFoxyGmFr 
So I managed to figure out a fix for the issue with previews not showing for JPEGs. I am using the docker image for my instance. The problem is that some images (especially from phones) include a broken embedded thumbnail in the EXIF metadata. FileBrowser tries to use that thumbnail, but if it’s zero-length, no preview appears. The issue seems to also affect videos, but as this was obsolete for my usecase I just debugged this for images.

Check if this issue is the case:
```
# Check thumbnail length for a single image
exiftool /path/to/image.jpg | grep -i Thumbnail
```
if you then see "Thumbnail Length : 0" you seem to have the same issue.

The solution is to remove only the embedded thumbnail while keeping all other metadata intact. You can do this on the host with exiftool:
```# Fix a single image
exiftool -ThumbnailImage= -overwrite_original /path/to/image.jpg

# Fix all JPEGs in a folder
exiftool -ThumbnailImage= -overwrite_original /path/to/folder/*.jpg

# Recursively fix all JPEGs in subfolders
exiftool -ThumbnailImage= -overwrite_original -r /path/to/folder/
```

To fix this issue at least somewhat decently I then added a cronjob for this command:

```
#Create the script
nano /home/user/docker/filebrowser/fix-jpeg-thumbnails.sh

#!/bin/bash
# Fix broken JPEG thumbnails in the FileBrowser folder
exiftool -ThumbnailImage= -overwrite_original -r /home/user/docker/filebrowser/files/

#Make it executable
chmod +x /home/user/docker/filebrowser/fix-jpeg-thumbnails.sh

#Create a cronjob for every 5 minutes
crontab -e

#Add this at the end
*/5 * * * * /home/user/docker/filebrowser/fix-jpeg-thumbnails.sh >/dev/null 2>&1

```

Regarding videos I suspect the container is lacking ffmpeg and therefore cannot create the previews. As the official image is a Alpine/Go container, its not that simple to add it. If I end up finding a fix for videos I will come back to this issue.

**Comment by z1haze** [comment]
Mine are all images from my daughters nice camera. Also these same images all have thumbnails on Google drive so I was hoping to was just a configuration issue. I guess the software is just buggy

**Comment by SaltyRiver** [comment]
Yes it seems to be a general issue with this project not being able to handle these kind of embedded thumbnails. That also explains why other services are able to display it correctly.

**Comment by ramiresviana** [comment]
> I am using the official docker image with a configuration very similar to the standard one. There is nothing in the logs that expands on the problem.
> 
> Here is an example of files and how they are displayed (these files differ only in EXIF):
> 
> ![Screenshot 2022-09-28 at 15 09 06](https://user-images.githubusercontent.com/33060302/192775057-d425155f-e08b-4945-9043-85c8f426b2a2.png)
> 
> The file "with_thumbnail.jpg" contains the above two lines, but the file "without_thumbnail.jpg" does not.
> 
> Files: [with_thumbnail.jpg](https://user-images.githubusercontent.com/33060302/192775486-d2a2891e-b8d1-412e-963c-468239ced46d.jpg), [without_thumbnail.jpg](https://user-images.githubusercontent.com/33060302/192775524-dfae198b-4a9d-4498-b434-fca60d9c56b0.jpg)
> 
> As I wrote above, I have already found the place where the mistake is most likely made and gave my solution. Hope for a speedy bug fix.

The issue is reproducible with the provided file sample.

**Comment by Turkey-Dinosaur** [comment]
I've just updated to the latest version ( v2.53.1 ) and thumbnails are now broken whereas they were not before i updated.

I'm running filebrowser via easypanel on a VPS.

Opened the terminal for the filebrowser project and ran **filebrowser config set --disableThumbnails=false**

filebrowser config cat shows:

Thumbnails Enabled: true

Issue still persists, no thumbnails displaying.

I've downgraded to each version until the thumbnails appear, the latest docker image version that loads thumbnails correctly for me is v2.48.0

**Comment by jerrywoo96** [comment]
https://github.com/filebrowser/filebrowser/discussions/5642

**Comment by hacdias** [comment]
File Browser is archived on 2026-09-01 and there will be no further releases, bug fixes, or security fixes, so I'm closing the remaining open issues rather than leaving them looking actionable. Thank you for taking the time to report it.
